### PR TITLE
add envFrom to container specs; sanitize comment with account info

### PR DESCRIPTION
**Pull request by jebbens**
**Is this a bug fix or adding new feature?**
Adding envFrom property to container specs in the controller, node, and values templates. Sanitize comment related to EKS IRSA in the serviceaccount-csi-controller template.

**What is this PR about? / Why do we need it?**
envFrom provides a simple way to reference configMaps across multiple resources. Sanitizing the comment in the template removes sensitive AWS account information and also makes the value more generic. 

**What testing is done?** 
Deployed the AWS EBS CSI driver using Helm with these changes made to the templates and referencing a configMap with values for http/s_proxy and no_proxy, and successfully ran a test using the dynamic provisioning example.

### CODE REVIEW

File: charts/aws-ebs-csi-driver/values.yaml [hunk 1]
```diff
@@ -77,6 +77,7 @@ controller:
   # If the default is not set and fstype is unset in the StorageClass, then no fstype will be set
   defaultFsType: ext4
   env: []
+  envFrom: []
```
Comment (torredil) [inline charts/aws-ebs-csi-driver/values.yaml:80]: Can you include a comment here for clarity? Something like "envFrom provides a way to reference configMaps across multiple resources".

**Review by torredil — COMMENTED** [review]
Can you bump up the Helm chart version [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/3f7f452e06b2babe95d0ef07153bd74a9cc34f48/charts/aws-ebs-csi-driver/Chart.yaml#L5) and update the [CHANGELOG](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/CHANGELOG.md)?

Rendered Helm chart, lgtm. Thanks for your contribution.

- `values.yaml`

```
controller:
  # If arbitrary args like "--aws-sdk-debug-log=true" need to be passed, use this value
  additionalArgs: []
  affinity: {}
  # The default filesystem type of the volume to provision when fstype is unspecified in the StorageClass.
  # If the default is not set and fstype is unset in the StorageClass, then no fstype will be set
  defaultFsType: ext4
  env: []
  envFrom:
  - configMapRef:
      name: env-configmap
  # If set, add pv/pvc metadata to plugin create requests as parameters.
  extraCreateMetadata: true

```

- `$ helm template ./ | grep -A 20 envFrom`

```
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: kubelet-dir
              mountPath: /var/lib/kubelet
              mountPropagation: "Bidirectional"
            - name: plugin-dir
              mountPath: /csi
            - name: device-dir
              mountPath: /dev
          ports:
            - name: healthz
              containerPort: 9808
              protocol: TCP
          livenessProbe:
            httpGet:
              path: /healthz
              port: healthz
            initialDelaySeconds: 10
            timeoutSeconds: 3
--
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: plugin-dir
              mountPath: /csi
            - name: registration-dir
              mountPath: /registration
        - name: liveness-probe
          image: k8s.gcr.io/sig-storage/livenessprobe:v2.5.0
          imagePullPolicy: IfNotPresent
          args:
            - --csi-address=/csi/csi.sock
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: plugin-dir
              mountPath: /csi
      volumes:
        - name: kubelet-dir
          hostPath:
            path: /var/lib/kubelet
            type: Directory
        - name: plugin-dir
          hostPath:
            path: /var/lib/kubelet/plugins/ebs.csi.aws.com/
            type: DirectoryOrCreate
        - name: registration-dir
          hostPath:
            path: /var/lib/kubelet/plugins_registry/
            type: Directory
        - name: device-dir
          hostPath:
--
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
          ports:
            - name: healthz
              containerPort: 9808
              protocol: TCP
          livenessProbe:
            httpGet:
              path: /healthz
              port: healthz
            initialDelaySeconds: 10
            timeoutSeconds: 3
            periodSeconds: 10
            failureThreshold: 5
          readinessProbe:
            httpGet:
              path: /healthz
--
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
        - name: csi-attacher
          image: k8s.gcr.io/sig-storage/csi-attacher:v3.4.0
          imagePullPolicy: IfNotPresent
          args:
            - --csi-address=$(ADDRESS)
            - --v=2
            - --leader-election=true
          env:
            - name: ADDRESS
              value: /var/lib/csi/sockets/pluginproxy/csi.sock
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
        - name: csi-resizer
          image: k8s.gcr.io/sig-storage/csi-resizer:v1.4.0
          imagePullPolicy: IfNotPresent
          args:
            - --csi-address=$(ADDRESS)
            - --v=2
            - --handle-volume-inuse-error=false
          env:
            - name: ADDRESS
              value: /var/lib/csi/sockets/pluginproxy/csi.sock
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: socket-dir
              mountPath: /var/lib/csi/sockets/pluginproxy/
        - name: liveness-probe
          image: k8s.gcr.io/sig-storage/livenessprobe:v2.5.0
          imagePullPolicy: IfNotPresent
          args:
            - --csi-address=/csi/csi.sock
          envFrom:
            - configMapRef:
                name: env-configmap
          volumeMounts:
            - name: socket-dir
              mountPath: /csi
      volumes:
        - name: socket-dir
          emptyDir: {}
```

File: charts/aws-ebs-csi-driver/values.yaml (hunk 1 above)
Comment (jebbens) [reply charts/aws-ebs-csi-driver/values.yaml:80]: Added "Use envFrom to reference ConfigMaps and Secrets across all containers in the deployment"

**Pull request merged**